src/d/parser/expression.d

```diff
@@ -1043,3 +1046,80 @@ StringLiteral parseStringLiteral(ref TokenRange trange) {
 	
 	return new StringLiteral(location, name.toString(trange.context));
 }
+
+/**
+ * Parse floating point literals
+ */
+FloatLiteral parseFloatLiteral(ref TokenRange trange) {
+	const location = trange.front.location;
+	auto litString = trange.front.toString(trange.context);
+
+	trange.match(TokenType.FloatLiteral);
+	import d.common.builtintype : BuiltinType;
+
+
+	assert(litString.length > 1);
+	//Look for a suffix
+	switch (litString[$-1])
+	{
+		// https://dlang.org/spec/lex.html#FloatSuffix
+		import std.conv : to;
+		case 'f':
+		case 'F':
+			const float f = litString[0..$-1].to!float;
+			return new FloatLiteral(location, f, BuiltinType.Float);
+		case 'L':
+			import source.exception;
+			throw new CompileException(location, "SDC does not support real literals yet");
+		default:
+			//Lexed correctly but no suffix, it's a double
+			const double d = litString[0..$].to!double;
+			return new FloatLiteral(location, d, BuiltinType.Double);
+	}
+}
+
+@("Test FloatLiteral parsing")
+unittest
+{
+	import source.context;
+	auto context = new Context();
+	import source.parserutil;
+	auto tokensFromString(string s) {
+		import source.name;
+		auto base = context.registerMixin(Location.init, s ~ '\0');
+		auto x = lex(base, context);
+		x.match(TokenType.Begin);
+		return x;
+	}
+	void floatRoundTrip(T)(const string floatString, const T floatValue)
+		if (__traits(isFloating, T))
+	{
+		import d.common.builtintype : BuiltinType;
+		const BuiltinType expectedType = is(T == float) ? BuiltinType.Float : BuiltinType.Double;
+		//Acceptable relativeError
+		const auto maxRelError = is(T == float) ? float.epsilon : double.epsilon;
+		auto tr = tokensFromString(floatString);
+		const result = parseFloatLiteral(tr);
+		import std.format : format;
+		const FloatLiteral fl = cast(FloatLiteral) result;
+		assert(fl, format("Got a %s from `%s`", typeid(result).toString(), floatString));
+
+		import std.math : isClose, abs;
```

Comment:
It's a bit bizare that abs is imported here and not in the if, where another import of std.math take place.

In general in this codebase, we don't specify symbols being imported, unless there is something to deambiguate. It gets verbose real quick, and the symbol list typically aren't maintained, so they just grow.

Reply:
Equally you want to avoid this style of import

```
import dmd.access;
import dmd.aggregate;
import dmd.aliasthis;
import dmd.arrayop;
import dmd.arraytypes;
import dmd.attrib;
import dmd.astcodegen;
import dmd.astenums;
import dmd.canthrow;
import dmd.chkformat;
import dmd.ctorflow;
import dmd.dscope;
import dmd.dsymbol;
import dmd.declaration;
import dmd.dclass;
import dmd.dcast;
import dmd.delegatize;
import dmd.denum;
import dmd.dimport;
import dmd.dinterpret;
import dmd.dmangle;
import dmd.dmodule;
import dmd.dstruct;
import dmd.dsymbolsem;
import dmd.dtemplate;
import dmd.errors;
import dmd.escape;
import dmd.expression;
import dmd.file_manager;
import dmd.func;
import dmd.globals;
import dmd.hdrgen;
import dmd.id;
import dmd.identifier;
import dmd.imphint;
import dmd.importc;
import dmd.init;
import dmd.initsem;
import dmd.inline;
import dmd.intrange;
import dmd.mtype;
import dmd.mustuse;
import dmd.nspace;
import dmd.opover;
import dmd.optimize;
import dmd.parse;
import dmd.printast;
import dmd.root.ctfloat;
import dmd.root.file;
import dmd.root.filename;
import dmd.common.outbuffer;
import dmd.root.rootobject;
import dmd.root.string;
import dmd.root.utf;
import dmd.semantic2;
import dmd.semantic3;
import dmd.sideeffect;
import dmd.safe;
import dmd.target;
import dmd.tokens;
import dmd.traits;
import dmd.typesem;
import dmd.typinf;
import dmd.utils;
import dmd.visitor;
```

Reply:
yes :P